test/test_cpp_extensions_jit.py

```diff
@@ -200,7 +200,21 @@ def test_jit_cuda_archflags(self):
             archflags["5.0;6.0+PTX;7.0;7.5"] = (['50', '60', '70', '75'], ['60'])
 
         for flags, expected in archflags.items():
-            self._run_jit_cuda_archflags(flags, expected)
+            try:
+                self._run_jit_cuda_archflags(flags, expected)
+            except RuntimeError as e:
+                # Using the device default (empty flags) may fail if the device is newer than the CUDA compiler
+                # This raises a RuntimeError with a specific message which we explictely ignore here
+                if not flags and "Error building" in str(e):
+                    pass
+                else:
+                    raise
+            try:
+                torch.cuda.synchronize()
+            except RuntimeError:
+                # Ignore any error, e.g. unsupported PTX code on current device
+                # to avoid errors from here leaking into other tests
+                pass
```

Comment:
This would slow down testing by a lot, can we do it only if exception was caught in the previous step

Also, instead of passing it would be good to raise `exception unittest.SkipTest("CUDA version mismatch")` or something like that

Reply:
I don't think it does slow down testing as nothing should be running on GPU at this point so this call is fast and done only a few times.

It cannot be done otherwise, check the description of the 2 individual commits and the PR:

- The first exception is raised only when the "device default" is not supported by the compiler.
- The exception here is raised when the device was not able to run the generated code, so the exact opposite of the first case.

IMO `skip` isn't appropriate here as the test was run and succeeded: `self._run_jit_cuda_archflags(flags, expected)` didn't raise any error, i.e. the JIT compilation worked. What we catch here is the execution error which is not what is being tested.   
Also you don't want to mark the whole test as skipped if a single configuration fails (more or less) expectantly, do you? 

Better would be to parametrize the test and maybe filter out stuff that isn't going to work anyway but that's quite some work and hence I proposed this PR to selectively ignore some exceptions.